### PR TITLE
Fixed get_*_list() tests

### PR DESCRIPTION
**Pull request by ddd-mtl**
removed ClientProtocol::FailureResult

### CODE REVIEW

File: crates/lib3h/src/gateway/gateway_dht.rs [hunk 1]
```diff
@@ -96,12 +97,36 @@ impl P2pGateway {
...
+            GatewayRequestToParent::Dht(payload),
+            Box::new(|me, response| {
+                trace!("Received requestEntry response in Gateway");
+                let dht_response = match response {
+                    GhostCallbackData::Response(Ok(GatewayRequestToParentResponse::Dht(d))) => d,
+                    GhostCallbackData::Response(Err(e)) => {
+                        panic!("Got error on GatewayRequest DHT: {:?} ", e);
+                    }
+                    GhostCallbackData::Timeout(_bt) => {
+                        // Not all requests expects a response so timeouts are expected
```
Comment (zippy) [inline crates/lib3h/src/gateway/gateway_dht.rs]: Would it make sense to distinguish between the ones that do and those that don't here then and panic on some of them?

Comment (ddd-mtl) [reply crates/lib3h/src/gateway/gateway_dht.rs]: I'll move the request within the RequestEntry variant and have the rest be a publish as before

File: crates/lib3h/tests/test_suites/two_get_lists.rs [hunk 2]
```diff
@@ -29,22 +27,21 @@ pub fn author_list_test(alex: &mut NodeMock, billy: &mut NodeMock) {
...
-    let fetch_data = unwrap_to!(srv_msg_list[0] => Lib3hServerProtocol::HandleFetchEntry);
+    let fetch_data = unwrap_to!(fetch_event => Lib3hServerProtocol::HandleFetchEntry);
+    println!("fetch_data: {:?}", fetch_data);
     // Respond
     alex.reply_to_HandleFetchEntry(&fetch_data)
         .expect("Reply to HandleFetchEntry should work");
-    let (did_work, _srv_msg_list) = alex.process().unwrap();
-    assert!(did_work);
 
     // Process the HoldEntry generated from receiving the HandleStoreEntryAspect
```
Comment (zippy) [inline crates/lib3h/tests/test_suites/two_get_lists.rs]: Is this comment correct?  "HoldEntry" no longer exits.

File: crates/lib3h/tests/test_suites/two_get_lists.rs [hunk 3]
```diff
@@ -133,93 +139,72 @@ pub fn many_aspects_test(alex: &mut NodeMock, billy: &mut NodeMock) {
...
+
+    let expected = "HandleStoreEntryAspect\\(StoreEntryAspectData \\{ request_id: \"[\\w\\d_~]+\", space_address: HashString\\(\"\\w+\"\\), provider_agent_id: HashString\\(\"billy\"\\), entry_address: HashString\\(\"entry_addr_1\"\\), entry_aspect: EntryAspectData \\{ aspect_address: HashString\\(\"[\\w\\d]+\"\\), type_hint: \"NodeMock\", aspect: \"hello-1\", publish_ts: \\d+ \\} \\}\\)";
+    let _results = assert2_msg_matches!(alex, billy, expected);
+    let mut entry = billy.get_entry(&ENTRY_ADDRESS_1).unwrap();
+    entry.aspect_list.sort();
+    assert_eq!(entry.aspect_list.len(), 3);
 
     // Send GossipingList
     // ==================
     // Send HoldingEntryList and should receive a HandleFetchEntry request from network module
```
Comment (zippy) [inline crates/lib3h/tests/test_suites/two_get_lists.rs]: comment old?

File: crates/lib3h/src/gateway/gateway_dht.rs [hunk 4]
```diff
@@ -96,7 +97,33 @@ impl P2pGateway {
...
+                            GhostCallbackData::Response(Err(e)) => {
+                                panic!("Got error on GatewayRequest DHT: {:?} ", e);
+                            }
+                            GhostCallbackData::Timeout(bt) => {
+                                panic!("Got timeout on GatewayRequest DHT: {:?}", bt);
+                            }
+                            _ => panic!("Got wrong response type"),
+                        };
+                        // #fullsync - received entry response after request from gossip list handling,
+                        // treat it as an erntry from author list handling.
```
Comment (neonphog) [inline crates/lib3h/src/gateway/gateway_dht.rs]:
```suggestion
                        // treat it as an entry from author list handling.
```

File: crates/lib3h/tests/utils/processor_harness.rs [hunk 5]
```diff
@@ -677,11 +677,19 @@ macro_rules! wait_engine_wrapper_did_work {
...
         loop {
             did_work = $crate::wait_engine_wrapper_did_work!($engine, false);
             if !did_work {
                 break;
             }
+            let elapsed = start.elapsed().expect("SystemTime failed");
+            if elapsed.as_secs() > 1 {
+                println!("elapsed = {:?}", elapsed.as_secs());
+                if elapsed.as_secs() > 60 {
+                    break;
```
Comment (neonphog) [inline crates/lib3h/tests/utils/processor_harness.rs]: : )

File: crates/lib3h/src/dht/mirror_dht.rs [hunk 6]
```diff
@@ -448,6 +453,15 @@ impl MirrorDht {
             // Ask for its data and broadcast it because we want fullsync.
             DhtRequestToChild::HoldEntryAspectAddress(entry) => {
                 trace!("DhtRequestToChild::HoldEntryAspectAddress: {:?}", entry);
+                // if its shallow, ask for actual data
+                if entry.aspect_list.len() > 0 && entry.aspect_list[0].aspect.len() == 0 {
```
Comment (neonphog) [inline crates/lib3h/src/dht/mirror_dht.rs]: Do we get shallow hold requests? I feel like we should make a different message type so we can assume which one is which and don't have to do a check like this.

File: crates/lib3h/src/gateway/gateway_dht.rs [hunk 7]
```diff
@@ -96,7 +97,33 @@ impl P2pGateway {
...
+                            GhostCallbackData::Response(Ok(
+                                GatewayRequestToParentResponse::Dht(d),
+                            )) => d,
+                            GhostCallbackData::Response(Err(e)) => {
+                                panic!("Got error on GatewayRequest DHT: {:?} ", e);
+                            }
+                            GhostCallbackData::Timeout(bt) => {
+                                panic!("Got timeout on GatewayRequest DHT: {:?}", bt);
+                            }
+                            _ => panic!("Got wrong response type"),
```
Comment (neonphog) [inline crates/lib3h/src/gateway/gateway_dht.rs]: I've been squishing all these failure cases into one arm for brevity / readability:

```rust
                        let dht_response = match response {
                            GhostCallbackData::Response(Ok(
                                GatewayRequestToParentResponse::Dht(d),
                            )) => d,
                            _ => panic!("invalid response type: {:?}", response),
                        }
```

I like how you are allowed to access the original variable in the catch-all arm if it fails to destructure : )

File: crates/lib3h/tests/utils/processor_harness.rs (hunk 5 above)
Comment (ddd-mtl) [reply crates/lib3h/tests/utils/processor_harness.rs]: :P

File: crates/lib3h/src/dht/mirror_dht.rs (hunk 6 above)
Comment (ddd-mtl) [reply crates/lib3h/src/dht/mirror_dht.rs]: > Do we get shallow hold requests? 
Yes from the gossiping list. Only mirror DHT needs to behave differently so I don't think we need a different message type.

**Pull request merged**